nemo/export/tensorrt_llm.py

```diff
@@ -524,7 +559,7 @@ def get_transformer_config(self, nemo_model_config):
             ffn_hidden_size=nemo_model_config.get('ffn_hidden_size'),
             layernorm_epsilon=nemo_model_config.get('layernorm_epsilon'),
             add_bias_linear=nemo_model_config.get('bias'),
-            num_moe_experts=nemo_model_config.get('num_moe_experts', None),
+            num_moe_experts=num_moe_experts if num_moe_experts > 0 else None,
             normalization=transformer_config_normalization,
             layernorm_zero_centered_gamma=layernorm_zero_centered_gamma,
```

Comment:
In https://github.com/NVIDIA/NeMo/pull/11614 there is also `gated_linear_unit=nemo_model_config.get('gated_linear_unit'),` added.

* Can we add it here? So that we merge only this MR to offload CI slightly.
* Can we write `nemo_model_config.get('gated_linear_unit', False)` to conform with the typing? There could be suprises maybe, see `None` vs `0` for `num_moe_experts`

Reply:
Added it in the [latest commit](https://github.com/NVIDIA/NeMo/pull/11409/commits/732e73568f430a17871eb14558bba149a30347e9)